src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,125 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.ExitCommand;
+import duke.helper.Parser;
+import duke.helper.Storage;
+import duke.helper.Ui;
+import duke.task.TaskList;
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+import java.io.File;
+
+public class Duke extends Application {
+
+    private String filePath;
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    /**
+     * Initializes Duke object.
+     */
+    public Duke() {
+        this.filePath = new File("").getAbsolutePath() + "/data/duke.txt";
+        ui = new Ui(null, userInput);
+        storage = new Storage(filePath, null);
```

Comment:
Consider using overloaded constructors to avoid passing null into Storage Constructor